### ISSUE TITLE
decompress_gzip_field Failed to decompress field in decompress_gzip_field processor: error decompressing field message: gzip: invalid header

### ISSUE THREAD

**Issue by zez3** [elastic/beats#40037]
I am trying to decompress a message with the https://www.elastic.co/guide/en/beats/filebeat/current/decompress-gzip-field.html
but it seems that it does not work. I get `error decompressing field gzip: invalid header`

The compression used for this field is unfortunately zlib forced by the following python lib:
https://github.com/keeprocking/pygelf/blob/master/pygelf/gelf.py#L85

and the decompress-gzip-field seems to only accept gzip
https://github.com/elastic/beats/blob/main/libbeat/processors/actions/decompress_gzip_field.go#L108

Please see the history here:
https://stackoverflow.com/questions/20762094/how-are-zlib-gzip-and-zip-related-what-do-they-have-in-common-and-how-are-they

would be nice for decompress_gzip_field processor to accept zlib as well
https://pkg.go.dev/compress/zlib

Love you

**Comment by pierrehilbert** [comment]
Thanks for sharing this enhancement suggestion.
Just to ensure we are on the same page here: you would like us to add zlib support in addition to gzip?
cc @nimarezainia

**Comment by zez3** [comment]
@pierrehilbert 

> Just to ensure we are on the same page here: you would like us to add zlib support in addition to gzip?

That is correct

**Comment by pierrehilbert** [comment]
I will let @nimarezainia, our PM have a look at this and we will check if this is something we would like to do and, if this is the case, when we will have time to spend on it.
Don't hesitate if you want to give it a try in the meantime, we will review your work with pleasure.

**Comment by zez3** [comment]
Does https://github.com/elastic/beats/pull/41584 fix this issue?
@cmacknz

**Comment by zez3** [comment]
> No, `decompress_gzip_field` still uses' compress/gzip and a standard gzip reader.

Cuz I saw in https://github.com/klauspost/compress

**Comment by zez3** [comment]
please reopen, I don't have permission to reopen this

**Comment by cmacknz** [comment]
Yeah using klauspost/compress would probably fix this, but we haven't made the switch in the place that helps this problem yet.

**Comment by nimarezainia** [comment]
@zez3  it's currently on the priority list for the roadmap, but we haven't allocated it time/resources due to other priorities. I hope soon.

**Comment by zez3** [comment]
Still valid
